crates/fuel-core/src/service.rs

```diff
@@ -132,6 +142,8 @@ impl FuelService {
             shutdown_listener,
         )?;
 
+        Self::write_metadata_at_genesis(&database)?;
```

Comment:
I think this logic should be inside of the `check_version` function. Maybe we need to rename `check_version` function to highlight that it can initialize the metadata